### PR TITLE
Use global::EDAnalyzers in GeneratorInterface/Core

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

The EDAnalyzers in GeneratorInterface/Core were either legacy or one::. GenXSecAnalyzer was prohibiting workflow 1361.181 from being able to use concurrent LuminosityBlocks efficiently.

Given that the two EDAnalyzers do no work on Event transitions (which is where we want the most thread-efficiency to be) using a std::mutex to only lock access at Lumi and Run transition will be fine.

As part of the change, I also removed the use of header files for the modules in the plugin directory.

#### PR validation:

The code compiles.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
FYI @makortel

**Comment by Dr15Jones** [comment]
please test workflow 1361.181

File: GeneratorInterface/Core/plugins/GenXSecAnalyzer.cc [hunk 1]
```diff
@@ -1,29 +1,144 @@
...
+
+    // LHE or Pythia/Herwig combined cross section of current luminosity block
+    // updated for each luminosity block, initialized in every run
+    // used for computation
+    CMS_THREAD_GUARD(GenXSecAnalyzer::mutex_) mutable std::map<int, GenLumiInfoProduct::XSec> currentLumiBlockLHEXSec_;
+  };
+}  // namespace gxsec
+
+class GenXSecAnalyzer
+    : public edm::global::EDAnalyzer<edm::RunCache<gxsec::RunCache>, edm::LuminosityBlockCache<gxsec::LumiCache>> {
```
Comment (makortel) [inline GeneratorInterface/Core/plugins/GenXSecAnalyzer.cc]: In https://github.com/cms-sw/cmssw/issues/29098 is was essentially concluded that this module does not have to be run in RelVals or in production. Now that the module is made thread-efficient, the issue becomes moot and I'll close it.

**Comment by qliphy** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Comment by silviodonato** [comment]
.

**Pull request merged**